src/components/shared/VirtualList.js

```diff
@@ -107,6 +107,15 @@ class VirtualListInnerChunk<Item> extends React.PureComponent<
           Math.max(visibleRangeStart, visibleRangeEnd)
         ).map(i => {
           const item = items[i];
+
+          // We compute isSpecial from the first position of item in the list,
+          // and the number of occurrences. Indeed we want to rerender the item
+          // when one of these properties changes.
+          const firstPosOfItem = specialItems.indexOf(item);
+          const countOfItem = specialItems.reduce(
+            (acc, specialItem) => (specialItem === item ? acc + 1 : acc),
+            0
+          );
```

Comment:
This change is because we want to rerender the row for the following cases:
* obviously one node moves from "non-selected-non-rightclicked" to one of "selected" or "rightclicked", or the opposite
* but also one node moving from "selected" to "right clicked", or more importantly from "right clicked" to "selected" (and they can be both in the same time).

That's why I changed the algorithm here: now `specialItem` is a string that's the concatenation of the first position of the item (-1, 0, or 1), and the number of times it's present in the array (currently 0, 1 or 2).

Reply:
I changed the name of `isSpecial` to something better and added some comments.